From 4af3320969f1a4a63ee37b72002f39278151f8fa Mon Sep 17 00:00:00 2001
From: Oneric <oneric@oneric.stub>
Date: Sun, 5 Jul 2026 00:00:00 +0000
Subject: [PATCH] parse: reject non-positive scales for vector clips

Matches guliverkli and xy-VSFilter.
We similarly prevent negative values for \p tags
but notably here the value must be at least one,
while \p tags allow zero values in both VSFilters
and existing libass.

Ref.: https://github.com/pinterf/xy-VSFilter/blob/4911cacb102bc0cbd41449793e54473265b04e62/src/subtitles/RTS.cpp#L2340
---
 libass/ass_parse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libass/ass_parse.c b/libass/ass_parse.c
index 28d49e71d8f2..e85fc5ebffdf 100644
--- a/libass/ass_parse.c
+++ b/libass/ass_parse.c
@@ -219,7 +219,7 @@ static bool parse_vector_clip(RenderContext *state,
 
     int scale = 1;
     if (nargs == 2)
-        scale = argtoi32(args[0]);
+        scale = FFMAX(argtoi32(args[0]), 1);
 
     struct arg text = args[nargs - 1];
     state->clip_drawing_text.str = text.start;
